prody/dynamics/adaptive.py

```diff
@@ -130,12 +130,17 @@ def calcStep(initial, target, n_modes, ensemble, defvecs, rmsds, mask=None, call
     c_sq = np.cumsum(np.power(normalised_overlaps, 2), axis=0)
 
     torf_Fmin = c_sq <= Fmin
-    if not np.any(torf_Fmin):
-        torf_Fmin[0] = True
-    
-    if not np.all(torf_Fmin):
-        i = np.where(torf_Fmin)[0].max()
-        torf_Fmin[i+1] = True
+
+    if Fmin == 0 and resetFmin:
```

Comment:
`torf_Fmin = c_sq <= Fmin` should be moved into the else block and in if block it initializes an all false one